doc/ipv6/ipv6_link_local.md

```diff
@@ -0,0 +1,378 @@
+| 0.1 | 08/02/2019  | Abhimanyu Devarapalli | Initial version                   |
+| 0.2 | 08/09/2019  | Abhimanyu Devarapalli | Addressed review comments         |
+| 0.3 | 08/20/2019  | Santosh Doke          | Addressed review comments         |
+
+# About this Manual
+This document provides general information about the IPv6 link-local enhancements in SONiC.
+# Scope
+This document describes the high level design of IPv6 link-local enhancements.
+
+# Definition/Abbreviation
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| ECMP                     | Equal Cost Multi Path               |
+| EUI                      | Extended Unique Identifier          |
+| swss                     | Switch State Service                |
+
+# 1 Feature Overview
+## 1.1 Requirements
+### 1.1.1 Functional Requirements
+1. Support auto-generation of IPv6 link-local address. An auto-generated IPv6 link-local address is an IPv6 unicast address that is automatically configured on interface. It uses the  prefix FE80::/64 and the interface identifier in the modified EUI-64 format. It is standardized through RFC 2373. Exactly one auto-generated IPv6 link-local address per interface is supported.
+2. Support manual configuration of IPv6 link-local address on the interface. For all practical use-case, only one link-local address per interface is required, in which case the number of link-local IPv6 addresses is limited by the number of router interfaces supported by the switching ASIC.
+3. Support IPv6 Link Local addresses on Ethernet, Port-channel, VLAN interfaces. Loopback interfaces do not require link-local address and hence do not have an auto-generated address.
+4. Support Neighbor Discovery  to determine/learn the IPv6 link-local addresses of neighbors on the attached links.
+5. Support IPv6 routes (including ECMP) via IPv6 link-local next hops. As SONIC uses same router MAC address for all L3 interfaces, it is mandatory to specify the interface along with the IPv6 link-local next hop.
+6. Support IPv6 protocols using Link local address.  This applies to all protocols, like BGP or ICMPv6, that support global IPv6 address.
+7. Support to trap control packets destined to IPv6 link-local address to CPU.
+8. Support filtering of packets with IPv6 link-local source or destination addresses. These packets must not be routed to other interfaces.  This implies utilities like trace route are not applicable for IPv6 link-local addresses. Also, ping to link-local address is only applicable for directly connected networks.
+9. Support BGP peering using unnumbered interface configuration. In this configuration, the IPv6 link-local address of the interface is used and the remote peer IPv6 link-local address is dynamically discovered to establish adjacency.
+10. IPv6 mode is disabled by default on an interface.

```

Comment:
As discussed, this can be changed to enabled as default and use some profile to disable